src/Sylius/Bundle/ApiBundle/StateProvider/Shop/OrderItem/ItemProvider.php

```diff
@@ -11,43 +11,41 @@
 
 declare(strict_types=1);
 
-namespace Sylius\Bundle\ApiBundle\DataProvider;
+namespace Sylius\Bundle\ApiBundle\StateProvider\Shop\OrderItem;
 
-use ApiPlatform\Core\DataProvider\ItemDataProviderInterface;
-use ApiPlatform\Core\DataProvider\RestrictedDataProviderInterface;
+use ApiPlatform\Metadata\Get;
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\State\ProviderInterface;
 use Sylius\Bundle\ApiBundle\Context\UserContextInterface;
-use Sylius\Component\Core\Model\AdminUserInterface;
+use Sylius\Bundle\ApiBundle\SectionResolver\ShopApiSection;
+use Sylius\Bundle\CoreBundle\SectionResolver\SectionProviderInterface;
 use Sylius\Component\Core\Model\CustomerInterface;
 use Sylius\Component\Core\Model\OrderItemInterface;
 use Sylius\Component\Core\Model\ShopUserInterface;
 use Sylius\Component\Core\Repository\OrderItemRepositoryInterface;
+use Webmozart\Assert\Assert;
 
-final class OrderItemItemDataProvider implements ItemDataProviderInterface, RestrictedDataProviderInterface
+/** @implements ProviderInterface<OrderItemInterface> */
+final readonly class ItemProvider implements ProviderInterface
```

Comment:
```suggestion
final readonly class GetProvider implements ProviderInterface
```
`operationName` + `Provider` WDYT?

Reply:
This also makes sense, but I don't have a strong opinion on which option is better, so it would be nice if more people commented 🤔 
Nevertheless, we will do a larger refactor (also of these existing providers) in a separate PR, so we will put the chosen version there as well